pkg/apis/core/v1/helper/helpers.go

```diff
@@ -370,3 +370,62 @@ func ScopedResourceSelectorRequirementsAsSelector(ssr v1.ScopedResourceSelectorR
 	selector = selector.Add(*r)
 	return selector, nil
 }
+
+// nodeSelectorRequirementsAsLabelRequirements converts the NodeSelectorRequirement
+// type to a labels.Requirement type.
+func nodeSelectorRequirementsAsLabelRequirements(nsr v1.NodeSelectorRequirement) (*labels.Requirement, error) {
```

Comment:
I don't remember where did I comment this, but I think that there are already helpers for this

> pkg/apis/core/helper/helpers.go:func NodeSelectorRequirementsAsSelector(nsm []core.NodeSelectorRequirement) (labels.Selector, error) {

Reply:
Addressed the comment here: https://github.com/kubernetes/kubernetes/pull/111123#discussion_r933120449